Math/NumberTheory/Primes/Factorisation/LinearAlgebra.hs

```diff
@@ -0,0 +1,112 @@
+module Math.NumberTheory.Primes.Factorisation.LinearAlgebra
+  ( SBVector(..)
+  , SBMatrix(..)
+  , testMatrix1
+  , testMatrix2
+  , mult
+  , linear --linearSolve
+  ) where
+
+import Data.Semigroup
+import System.Random
+import Debug.Trace
+import qualified Data.List as L
+import qualified Data.Vector as V
+import qualified Data.Vector.Unboxed as U
+import qualified Data.IntSet as S
+import qualified Data.IntMap as I
+import System.IO.Unsafe
+import GHC.Clock
+import Data.Bit
+import qualified Data.Semiring as S
+
+-- Sparse Binary Vector
+newtype SBVector = SBVector
+  { set :: S.IntSet
+  } deriving (Show)
+
+-- Vectors form a group (in particular a monoid) under addition.
+instance Semigroup SBVector where
+  SBVector v1 <> SBVector v2 = SBVector ((v1 S.\\ v2) <> (v2 S.\\ v1))
+
+instance Monoid SBVector where
+  mempty = SBVector mempty
+
+-- Sparse Binary Matrix
+newtype SBMatrix = SBMatrix (I.IntMap SBVector) deriving (Show)
+
+mult :: SBMatrix -> SBVector -> SBVector
+mult (SBMatrix matrix) (SBVector vector) = foldMap (matrix I.!) (S.toList vector)
+
+-- Multiplication of matrices
+instance Num SBMatrix where
+  (+) (SBMatrix m1) (SBMatrix m2) = SBMatrix (I.unionWith (<>) m1 m2)
+  (*) m1 (SBMatrix m2) = SBMatrix (I.map (m1 `mult`) m2)
+  negate = id
+
+instance Semigroup SBMatrix where
+  (<>) = (*)
+
+size :: SBMatrix -> Int
+size (SBMatrix m) = I.size m
+
+randomSublist :: [Int] -> StdGen -> [Int]
+randomSublist list gen = fmap fst $ filter snd $ zip list (randoms gen)
+
+-- linearSolve :: SBMatrix -> SBVector
```

Comment:
Why are you commiting with unused code?

Reply:
Just to clarify the process: I asked @folidota to dump a snapshot of his work every day, so sometimes he has to comment out unfinished work just to get everything else compiling. I'm fine with it as long as this PR is marked as a draft.